engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -3516,6 +3517,16 @@ private void checkPlatformCompatible(AppPackage ap)
     }
   }
 
+  private void checkJavaVersionCompatible(AppPackage ap)
+  {
+    String buildJdkVersion = ap.getBuildJdkVersion();
+    String systemVersion = System.getProperty("java.version");
+    if (VersionInfo.compare(buildJdkVersion, systemVersion) > 0) {
```

Comment:
GetAppPackageInfoCommand prints all values of a JsonSerialization of the AppPackage class. Since I added buildJdk as a member in AppPackage it will already print it.